blazar/tests/enforcement/test_enforcement.py

```diff
@@ -0,0 +1,209 @@
+        'local_gb': 10,
+    }
+
+
+def get_fake_lease(**kwargs):
+    fake_lease = {
+        'id': '1',
+        'name': u'lease_test',
+        'start_date': u'2014-01-01 01:23',
+        'end_date': u'2014-02-01 13:37',
+        'user_id': 111,
+        'project_id': 222,
+        'trust_id': u'35b17138b3644e6aa1318f3099c5be68',
+        'reservations': [{u'resource_id': u'1234',
+                          u'resource_type': u'virtual:instance'}],
+        'events': [],
+        'before_end_date': u'2014-02-01 10:37',
+        'action': None,
+        'status': None,
+        'status_reason': None}
+
+    if kwargs:
+        fake_lease.update(kwargs)
+
+    return fake_lease
+
+
+class FakeFilter(filters.base_filter.BaseFilter):
+
+    enforcement_opts = [
+        cfg.IntOpt('fake_opt', default=1, help='This is a fake config.'),
+    ]
+
+    def __init__(self, conf=None):
+        super(FakeFilter, self).__init__(conf=conf)
+
+    def check_create(self, context, lease_values):
+        pass
+
+    def check_update(self, context, current_lease_values, new_lease_values):
+        pass
+
+    def on_end(self, context, lease_values):
+        pass
+
+
+@ddt.ddt
+class EnforcementTestCase(tests.TestCase):
+    def setUp(self):
+        super(EnforcementTestCase, self).setUp()
+
+        self.cfg = cfg
+        self.region = 'RegionOne'
+        self.all_filters = ['FakeFilter']
+        self.filters = filters
+
+        self.enforcement = enforcement.UsageEnforcement()
+        self.filters.FakeFilter = FakeFilter
+        cfg.CONF.set_override(
+            'enabled_filters', self.all_filters, group='enforcement')
+        cfg.CONF.set_override('os_region_name', self.region)
+
+        self.enforcement.load_filters()
+        self.fake_service_catalog = [
+            dict(
+                type='identity', endpoints=[
+                    dict(
+                        interface='public', region=self.region,
+                        url='https://fakeauth.com')
+                ]
+            )
+        ]
+
+        self.ctx = context.BlazarContext(
+            user_id=111, project_id=222,
+            service_catalog=self.fake_service_catalog)
+        self.set_context(self.ctx)
+
+        self.fake_host_id = '1'
+        self.fake_host = {
+            'id': self.fake_host_id,
+            'hypervisor_hostname': 'hypvsr1',
+            'service_name': 'compute1',
+            'vcpus': 4,
+            'cpu_info': 'foo',
+            'hypervisor_type': 'xen',
+            'hypervisor_version': 1,
+            'memory_mb': 8192,
+            'local_gb': 10,
+        }
+
+    def tearDown(self):
+        super(EnforcementTestCase, self).tearDown()
+
+    def test_load_filters(self):
+        print(self.enforcement.enabled_filters)
+        self.assertEqual(len(self.enforcement.enabled_filters), 1)
+
+        fake_filter = self.enforcement.enabled_filters[0]
+
+        self.assertEqual(fake_filter.conf.enforcement.fake_opt, 1)
```

Comment:
You could also do a type check, `self.assertIsInstance(fake_filter, FakeFilter)`. I'd maybe also check that the length of the list is 1.

Reply:
Good call. Updated.